glide-core/src/client/value_conversion.rs

```diff
@@ -67,9 +67,78 @@ pub(crate) fn convert_to_expected_type(
             Value::Nil => Ok(value),
             _ => Ok(Value::Double(from_owned_redis_value::<f64>(value)?.into())),
         },
+        ExpectedReturnType::ArrayOfDouble => match value {
+            Value::Array(array) => {
+                let mut result = Vec::new();
+                for element in array {
+                    match element {
+                        Value::BulkString(_) => {
+                            result.push(Value::Double(
+                                from_owned_redis_value::<f64>(element)?.into(),
+                            ));
+                        }
+                        _ => {
+                            result.push(element);
+                        }
+                    }
+                }
+                Ok(Value::Array(result))
+            }
+            _ => Err((
+                ErrorKind::TypeError,
+                "Response couldn't be converted to Array",
+                format!("(response was {:?})", value),
+            )
+                .into()),
+        },
     }
 }
 
+fn convert_array_to_map(
+    array: Vec<Value>,
+    expected_return_type: &ExpectedReturnType,
+) -> RedisResult<Value> {
+    let mut map = Vec::new();
+    let mut iterator = array.into_iter();
+    while let Some(key) = iterator.next() {
+        match key {
+            Value::Array(inner_array) => {
+                if inner_array.len() != 2 {
+                    return Err((
+                        ErrorKind::TypeError,
+                        "Array inside map must contain exactly two elements",
+                    )
+                        .into());
+                }
+                let mut inner_iterator = inner_array.into_iter();
+                let inner_key = inner_iterator.next().unwrap();
+                let inner_value = inner_iterator.next().unwrap();
+                match expected_return_type {
+                    ExpectedReturnType::MapOfStringToDouble => {
+                        map.push((
+                            inner_key,
+                            Value::Double(from_owned_redis_value::<f64>(inner_value)?.into()),
+                        ));
+                    }
+                    _ => {
+                        map.push((inner_key, inner_value));
+                    }
+                }
+            }
+            _ => {
+                let Some(value) = iterator.next() else {
+                    return Err((
+                        ErrorKind::TypeError,
+                        "Response has odd number of items, and cannot be entered into a map",
+                    )
+                        .into());
+                };
+                map.push((key, value));
+            }
+        }
+    }
+    Ok(Value::Map(map))
+}
 pub(crate) fn expected_type_for_cmd(cmd: &Cmd) -> Option<ExpectedReturnType> {
     if cmd.arg_idx(0) == Some(b"ZADD") {
```

Comment:
instead of all of these `if`s, we should use a `match` statement, either on `cmd.arg_idx(0)` or on `command`.